From f3ab017a229c9eca400a55d24cd9a4f759c4baa8 Mon Sep 17 00:00:00 2001
From: Stephen Finucane <stephenfin@redhat.com>
Date: Sat, 29 Aug 2026 11:27:47 +0100
Subject: [PATCH] json-rpc: Rename TLS options

These options control TLS configuration for the JSON RPC server.
keystoneauth 5.17.0 adds new, identically-named options that control TLS
configuration for the *client*. Allow users to distinguish between the
two by adding a `server_` prefix to the former. Aliases are provided,
though these introduce a small upgrade impact: users who had overridden
these options to control server configuration will see the same
configuration inherited for clients (or rather, will as soon as ksa is
bumped). This will be covered via an update to the release note in a
subsequent change.

Change-Id: I1972c85de47fc70758753665488f822b4320a264
Signed-off-by: Stephen Finucane <stephenfin@redhat.com>
---
 ironic/common/wsgi_service.py                 |  4 ++--
 ironic/conf/api.py                            |  6 +++--
 ironic/conf/json_rpc.py                       |  6 +++--
 ironic/tests/unit/common/test_json_rpc.py     | 22 ++++++++---------
 ironic/tests/unit/common/test_wsgi_service.py | 24 +++++++++----------
 ...me-json-rpc-tls-opts-88508aa569381c4d.yaml |  7 ++++++
 6 files changed, 40 insertions(+), 29 deletions(-)
 create mode 100644 releasenotes/notes/rename-json-rpc-tls-opts-88508aa569381c4d.yaml

diff --git a/ironic/common/wsgi_service.py b/ironic/common/wsgi_service.py
index b624916325..1e004fb92c 100644
--- a/ironic/common/wsgi_service.py
+++ b/ironic/common/wsgi_service.py
@@ -92,14 +92,14 @@ def __init__(self, name, app, conf, use_ssl=None):

             validate_cert_paths(cert_file, key_file)

-            tls_ciphers = conf.tls_ciphers
+            tls_ciphers = conf.server_tls_ciphers
             self.server.ssl_adapter = cheroot_ssl.BuiltinSSLAdapter(
                 certificate=cert_file,
                 private_key=key_file,
                 ciphers=tls_ciphers,
             )

-            tls_min = conf.tls_minimum_version
+            tls_min = conf.server_tls_minimum_version
             if tls_min:
                 tls_utils.check_tls_version_supported(tls_min)
                 version = tls_utils.TLS_VERSION_MAP[tls_min]
diff --git a/ironic/conf/api.py b/ironic/conf/api.py
index 0b9fe5c9c5..567c74138c 100644
--- a/ironic/conf/api.py
+++ b/ironic/conf/api.py
@@ -121,7 +121,8 @@ def __call__(self, value):
                help="Private key file to use when starting "
                     "the server securely."),
     cfg.StrOpt(
-        'tls_minimum_version',
+        'server_tls_minimum_version',
+        deprecated_name='tls_minimum_version',
         default='1.3',
         choices=[
             ('1.2',
@@ -141,7 +142,8 @@ def __call__(self, value):
                "external service such as a reverse proxy "
                "or a WSGI application container.")),
     cfg.StrOpt(
-        'tls_ciphers',
+        'server_tls_ciphers',
+        deprecated_name='tls_ciphers',
         help=_("Set the list of available ciphers for the "
                "integrated API server. The value should be "
                "a string in the OpenSSL cipher list format. "
diff --git a/ironic/conf/json_rpc.py b/ironic/conf/json_rpc.py
index 10c97556e3..5a44da3f6c 100644
--- a/ironic/conf/json_rpc.py
+++ b/ironic/conf/json_rpc.py
@@ -49,7 +49,8 @@
     cfg.StrOpt('key_file',
                help=_("Private key file matching cert_file.")),
     cfg.StrOpt(
-        'tls_minimum_version',
+        'server_tls_minimum_version',
+        deprecated_name='tls_minimum_version',
         default='1.3',
         choices=[
             ('1.2',
@@ -66,7 +67,8 @@
                "TLS 1.3 since TLS 1.0 and 1.1 are "
                "deprecated (RFC 8996).")),
     cfg.StrOpt(
-        'tls_ciphers',
+        'server_tls_ciphers',
+        deprecated_name='tls_ciphers',
         help=_("Set the list of available ciphers for the "
                "JSON-RPC server. The value should be a "
                "string in the OpenSSL cipher list format. "
diff --git a/ironic/tests/unit/common/test_json_rpc.py b/ironic/tests/unit/common/test_json_rpc.py
index 04c7effe6a..051496ca59 100644
--- a/ironic/tests/unit/common/test_json_rpc.py
+++ b/ironic/tests/unit/common/test_json_rpc.py
@@ -399,7 +399,7 @@ def test_json_rpc_tls_minimum_version(
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_minimum_version='1.3',
+        self.config(server_tls_minimum_version='1.3',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -430,7 +430,7 @@ def test_json_rpc_tls_ciphers(self, mock_ssl_adapter,
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_ciphers='ECDHE+AESGCM',
+        self.config(server_tls_ciphers='ECDHE+AESGCM',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -462,9 +462,9 @@ def test_json_rpc_tls_all_options(
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_minimum_version='1.3',
+        self.config(server_tls_minimum_version='1.3',
                     group='json_rpc')
-        self.config(tls_ciphers='ECDHE+AESGCM',
+        self.config(server_tls_ciphers='ECDHE+AESGCM',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -495,7 +495,7 @@ def test_json_rpc_tls_minimum_version_1_2(
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_minimum_version='1.2',
+        self.config(server_tls_minimum_version='1.2',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -547,9 +547,9 @@ def test_json_rpc_tls_options_ignored_when_ssl_disabled(
             self, mock_ssl_adapter):
         """TLS hardening options have no effect with SSL off."""
         self.config(use_ssl=False, group='json_rpc')
-        self.config(tls_minimum_version='1.3',
+        self.config(server_tls_minimum_version='1.3',
                     group='json_rpc')
-        self.config(tls_ciphers='ECDHE+AESGCM',
+        self.config(server_tls_ciphers='ECDHE+AESGCM',
                     group='json_rpc')

         server.WSGIService(
@@ -571,7 +571,7 @@ def test_json_rpc_tls_min_version_logged(
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_minimum_version='1.3',
+        self.config(server_tls_minimum_version='1.3',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -600,7 +600,7 @@ def test_json_rpc_tls_ciphers_logged(
                     group='json_rpc')
         self.config(key_file='/path/to/key',
                     group='json_rpc')
-        self.config(tls_ciphers='ECDHE+AESGCM',
+        self.config(server_tls_ciphers='ECDHE+AESGCM',
                     group='json_rpc')

         mock_context = mock.Mock()
@@ -628,9 +628,9 @@ def test_json_rpc_ssl_legacy_group_with_tls_options(
             mock_validate):
         """TLS options work with legacy [ssl] cert path."""
         self.config(use_ssl=True, group='json_rpc')
-        self.config(tls_minimum_version='1.3',
+        self.config(server_tls_minimum_version='1.3',
                     group='json_rpc')
-        self.config(tls_ciphers='ECDHE+AESGCM',
+        self.config(server_tls_ciphers='ECDHE+AESGCM',
                     group='json_rpc')
         from oslo_service import sslutils
         sslutils.register_opts(cfg.CONF)
diff --git a/ironic/tests/unit/common/test_wsgi_service.py b/ironic/tests/unit/common/test_wsgi_service.py
index 72862897ca..6c8a91284c 100644
--- a/ironic/tests/unit/common/test_wsgi_service.py
+++ b/ironic/tests/unit/common/test_wsgi_service.py
@@ -124,7 +124,7 @@ def test_ssl_with_tls_minimum_version(self, mock_validate_tls,
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_minimum_version='1.2', group='api')
+        self.config(server_tls_minimum_version='1.2', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -155,7 +155,7 @@ def test_ssl_with_tls_ciphers(self, mock_validate_tls,
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_ciphers='ECDHE+AESGCM', group='api')
+        self.config(server_tls_ciphers='ECDHE+AESGCM', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -187,8 +187,8 @@ def test_ssl_with_all_tls_options(
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_minimum_version='1.3', group='api')
-        self.config(tls_ciphers='ECDHE+AESGCM', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')
+        self.config(server_tls_ciphers='ECDHE+AESGCM', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -244,8 +244,8 @@ def test_tls_options_ignored_when_ssl_disabled(
             self, mock_server):
         """TLS hardening options have no effect with SSL off."""
         self.config(enable_ssl_api=False, group='api')
-        self.config(tls_minimum_version='1.3', group='api')
-        self.config(tls_ciphers='ECDHE+AESGCM', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')
+        self.config(server_tls_ciphers='ECDHE+AESGCM', group='api')

         mock_server.return_value = self.server

@@ -274,8 +274,8 @@ def test_ssl_legacy_group_with_tls_options(
             mock_ssl_adapter, mock_server):
         """TLS options work with legacy [ssl] cert path."""
         self.config(enable_ssl_api=True, group='api')
-        self.config(tls_minimum_version='1.3', group='api')
-        self.config(tls_ciphers='ECDHE+AESGCM', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')
+        self.config(server_tls_ciphers='ECDHE+AESGCM', group='api')
         self.config(cert_file='/path/to/cert', group='ssl')
         self.config(key_file='/path/to/key', group='ssl')

@@ -309,7 +309,7 @@ def test_ssl_tls_min_version_logged(
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_minimum_version='1.3', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -339,7 +339,7 @@ def test_ssl_tls_version_unsupported_compile_time(
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_minimum_version='1.3', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -364,7 +364,7 @@ def test_ssl_tls_version_exceeds_crypto_policy(
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_minimum_version='1.3', group='api')
+        self.config(server_tls_minimum_version='1.3', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
@@ -400,7 +400,7 @@ def test_ssl_tls_ciphers_logged(
         self.config(enable_ssl_api=True, group='api')
         self.config(cert_file='/path/to/cert', group='api')
         self.config(key_file='/path/to/key', group='api')
-        self.config(tls_ciphers='ECDHE+AESGCM', group='api')
+        self.config(server_tls_ciphers='ECDHE+AESGCM', group='api')

         mock_server.return_value = self.server
         mock_context = mock.Mock()
diff --git a/releasenotes/notes/rename-json-rpc-tls-opts-88508aa569381c4d.yaml b/releasenotes/notes/rename-json-rpc-tls-opts-88508aa569381c4d.yaml
new file mode 100644
index 0000000000..055c17346c
--- /dev/null
+++ b/releasenotes/notes/rename-json-rpc-tls-opts-88508aa569381c4d.yaml
@@ -0,0 +1,7 @@
+---
+deprecations:
+  - |
+    The ``[json_rpc] tls_minimum_version`` and ``[json_rpc] tls_ciphers``
+    config options have been renamed to ``[json_rpc]
+    server_tls_minimum_version`` and ``[json_rpc] server_tls_ciphers``,
+    respectively, to distinguish them from new client-only options.
